src/cdk/overlay/overlay-directives.ts

```diff
@@ -425,6 +435,13 @@ export class CdkConnectedOverlay implements OnDestroy, OnChanges {
   static ngAcceptInputType_push: BooleanInput;
 }
 
+// Checks whether the provided object is CdkOverlayOrigin or not.
+// We are not using `instanceof` to be backward compatible.
+// For example, some existing clients may be sending an `{elementRef: Element}`
+// instead of `CdkOverlayOrigin` to achieve the same behavior.
+function isCdkOverlayOrigin(origin: unknown): origin is CdkOverlayOrigin {
```

Comment:
I did this to be backward compatible.
I actually started with `instanceof` check. But then I saw `mdc-select` tests are failing.
The reason of the failure is to mimic `CdkOverlayOrigin`, `mdc-select` was setting an object as input instead of a `CdkOverlayOrigin`.
I fixed `mdc-select`, but there can be other cases in the wild which are doing this hack.
Should I revert back to `instanceof` ?

Reply:
I don't think that the select case is something we support explicitly, it just something that I took as a shortcut.

Reply:
Done. Thanks Kristiyan for the reviews :smiley: